File: gcp-go-fn/function.go
package p

import (
	"encoding/json"
	"fmt"
	"html"
	"net/http"
)

func enableCors(w *http.ResponseWriter) {
    (*w).Header().Set("Access-Control-Allow-Origin", "https://furikuri.net")
    (*w).Header().Set("Access-Control-Allow-Methods", "GET, POST")
    (*w).Header().Set("Access-Control-Allow-Headers", "Content-Type")
}

func HelloWorld(w http.ResponseWriter, r *http.Request) {
	var d struct {
		Name string `json:"name"`
	}
    var name = "World";
	json.NewDecoder(r.Body).Decode(&d);
    
	if d.Name != "" {
		name = d.Name
	}
    enableCors(&w)
	fmt.Fprint(w, html.EscapeString("Hello " + name + "! From GCP + Go"))
}

File: aws-python-fn/function.py
import json

def lambda_handler(event, context):
    try:
        name = json.loads(event['body'])['name']
    except:
        name = 'World'

    if not name:
        name = 'World'

    headers = {
        'Content-Type': 'text/plain',
        'Access-Control-Allow-Origin': 'https://furikuri.net',
        'Access-Control-Allow-Methods': 'GET, POST',
        'Access-Control-Allow-Headers': 'Content-Type'
    }

    return {
        'statusCode': 200,
        'headers': headers,
        'body': 'Hello ' + name + '! From AWS + Python'
    }
